### ISSUE TITLE
END doesn't capture the environment

### ISSUE THREAD

**Issue by chrisseaton** [jruby/jruby#4257]
```
$ ruby -e 'x = 14 ; END { p x }'
14
$ bin/jruby -e 'x = 14 ; END { p x }'
nil
$ ~/.rbenv/versions/rbx-2.11/bin/ruby -e 'x = 14 ; END { p x }'
14
```

`PostExeParseNode#getScope()` is null when it comes out of the parser. It should be the top level scope I think.

There's some musings here about the limitations of the implementation of `END` that might relevant.

https://github.com/jruby/jruby/blob/bc7c127c62af69d896572f7b20c4f3abbafd2b6f/core/src/main/java/org/jruby/ir/IRBuilder.java#L3299-L3299

I don't think dropping support for `END` is a good idea though - it's used at least in `mkmf` and `profile`.

**Comment by chrisseaton** [comment]
See 1c18c37a61d9f5b3f70b04a732991abe873c518e

**Comment by enebo** [comment]
@chrisseaton yeah don't take that musing seriously.  You are also correct END does not make it's own variable scope  (you can tell too if you invoke local_variables in script and in END.  I want to mention one other not well known thing:

```ruby
END { p 1 }
END { p 2 }
```
This will print 2 then 1 since end blocks are executed last to first.

**Comment by chrisseaton** [comment]
Closing due to inactivity.

**Comment by enebo** [comment]
This is clearly still not fixed.  Re-opening.
